lisa/sut_orchestrator/hyperv/schema.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from typing import List
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+
+
+@dataclass_json
+@dataclass
+class HypervServer:
+    address: str
+    username: str
+    password: str
+
+
+@dataclass_json
+@dataclass
+class ExtraArgs:
+    command: str
+    args: str
+
+
+@dataclass_json()
+@dataclass
+class VHDSourceSchema(schema.TypedSchema, schema.ExtendableSchemaMixin):
+    pass
+
+
+@dataclass_json()
+@dataclass
+class LocalVHDSourceSchema(VHDSourceSchema):
+    vhd_path: str = ""
+    extract: bool = False
+
+
+@dataclass_json
+@dataclass
+class HypervPlatformSchema:
+    vhd_source: VHDSourceSchema
```

Comment:
`build`

Reply:
That name doesn't make sense here. The names in the schema should be such that its use is clear and obvious to the user. Naming it "vhd_source" makes it obvious to the user that they are supposed to specify the VHD details here. Simply calling it "build" gives no such hint to the user and is confusing.

Reply:
ok, we can call it source, the schema is like below. The source object should provide a method to query all destination files together.
 
```yaml
source
  type: 
  files:
    - source: # absolute or relative path, depends on implementation. It can be a name pattern or folder for multiple files.
      destination: # if it's relative path, the root is lisa working path.
      overwrite: true/false
      unzip: true/false
```

Reply:
1. Is destination optional? For things like VHD, VMGS etc. destination should not be needed. The platform will manage the destination.
2. What values can `type` take? Will it be `local`, `ado`, `azure_storage` etc or something else?
3. From the above schema it looks like multiple files can be specified under `files`. If so, how will the platform know which one is VHD, VMGS or some other file?
4. I didn't quite understand this part: "source object should provide a method to query all destination files together". Do you have an example?

Reply:
> The whole `source` object is optional, if vhd paths are provided in the ImageSchema.

Okay, so if I want to specify a local vhd path, I don't need a `source`. I can specify in `ImageSchema`. But if I want to specify a local `.zip` file that contains the VHD, I would need to specify a `source` with `type: local` & `unzip: true` in the runbook. Right?

What happens if both are specified i.e. vhd path in ImageSchema and `source`? Should one take precedence, or should we error out?

Reply:
> What happens if both are specified i.e. vhd path in ImageSchema and source? Should one take precedence, or should we error out?

If vhd path is specified, it should be able to find from the source files. It's used to be clearer on which file is taken, but not required. If not, LISA should fail to see the vhd is not found from source.

Reply:
Please see the latest changes.